generic.go

```diff
@@ -630,3 +630,20 @@ func SkipUntil[V any](slice []V, matcher AnyMatcher) []V {
 func SkipWhile[V any](slice []V, matcher AnyMatcher) []V {
 	return SkipUntil(slice, Not(matcher))
 }
+
+// Sliding returns a "sliding window" view of the items in `slice`
+func Sliding[V any](slice []V, windowSize int) [][]V {
+	if windowSize < 1 || len(slice) == 0 {
+		return [][]V{}
```

Comment:
I think returning `nil` here is more appropriate, considering it would have the same effect but also allowing for the iDiOmAtIc `if Slinding(...) == nil`